### DISCUSSION TITLE
mise doctor command does not have an option not to check of newer version

### DISCUSSION THREAD

**Discussion by ssbarnea** [jdx/mise#3874] [General]
I tried to use `mise doctor` on CI in order to ensure that the newly installed mise was setup correctly by https://github.com/jdx/mise-action/tree/v2/ but I discovered that it was failing reporting an outdated miss,... bit ironic considering that was installed few seconds before.

```
new mise version 2024.12.22 available, currently on 2024.12.21
```

I think that we need a switch or env variable that does disable to version check, especially as an outdated version changes the return code of `mise doctor`.

I did a brief grep on the source code and apparently there is no undocumented way to do it.

**Comment by jdx** [comment]
I'd accept a PR to move this to a warning, there are already warning and error checks so it should be simple

**Comment by Marukome0743** [comment]
This has since been implemented in [#5317](https://github.com/jdx/mise/pull/5317) and released in mise v2025.6.3.

An available mise update is now recorded as a warning rather than a doctor error. `mise doctor` may still display:

```text
new mise version ... available, currently on ...
```

but that warning alone no longer causes a non-zero exit code. The command only exits with status 1 when an actual doctor error is found.

This means `mise doctor` can now be used in CI without failing solely because a newer mise release appeared immediately after installation.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
